v2/dtos/resourceoperation.go

```diff
@@ -0,0 +1,32 @@
+//
+// Copyright (C) 2020 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package dtos
+
+import "github.com/edgexfoundry/go-mod-core-contracts/v2/models"
+
+// ResourceOperation defines an operation of which a device is capable
+// This object and its properties correspond to the ResourceOperation object in the APIv2 specification:
+// https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-metadata/2.x#/ResourceOperation
+type ResourceOperation struct {
+	Operation      string            `json:"operation" yaml:"operation,omitempty"`
+	DeviceResource string            `json:"deviceResource" yaml:"deviceResource,omitempty"` // The replacement of Object field
+	Parameter      string            `json:"parameter" yaml:"parameter,omitempty"`
+	DeviceCommand  string            `json:"deviceCommand" yaml:"deviceCommand,omitempty"` // The replacement of Resource field
+	Secondary      []string          `json:"secondary" yaml:"secondary,omitempty"`
+	Mappings       map[string]string `json:"mappings" yaml:"mappings,omitempty"`
```

Comment:
Add validation as appropriate for Add operation

Reply:
@lenny-intel @tonyespy @jpwhitemn @iain-anderson 
Should we keep the **DeviceCommand**  and **Secondary** field?  
What is the usage of **Secondary**?

Reply:
Intention is still to implement DeviceCommand at some point. I think not for Secondary though, it could be removed if all agree.

Reply:
does the DeviceCommand here mean chained command?

Reply:
yes

Reply:
@iain-anderson , can we remove both and add them back when/if needed?

Reply:
Removed. Agree to remove both and add them back when/if needed.